src/gui/preferences.c

```diff
@@ -2335,6 +2335,125 @@ GtkWidget *dt_gui_preferences_int(GtkGrid *grid, const char *key)
   return w;
 }
 
+static void
+_gui_preferences_enum_callback(GtkWidget *widget, gpointer data)
+{
+  GtkTreeIter iter;
+  if(gtk_combo_box_get_active_iter(GTK_COMBO_BOX(widget), &iter))
+  {
+    gchar *s = NULL;
+    gtk_tree_model_get(gtk_combo_box_get_model(GTK_COMBO_BOX(widget)), &iter, 0, &s, -1);
+    dt_conf_set_string((char *)data, s);
+    g_free(s);
+  }
+}
+
+void _gui_preferences_enum_set(GtkWidget *widget, const char *str)
+{
+  GtkTreeModel *model = gtk_combo_box_get_model(GTK_COMBO_BOX(widget));
+  GtkTreeIter iter;
+  gboolean valid = gtk_tree_model_get_iter_first(model, &iter);
+  gint i = 0;
+  gboolean found = FALSE;
+  while(valid)
+  {
+    char *value;
+    gtk_tree_model_get(model, &iter, 0, &value, -1);
+    if(!g_strcmp0(value, str))
+    {
+      g_free(value);
+      found = TRUE;
+      break;
+    }
+    i++;
+    g_free(value);
+    valid = gtk_tree_model_iter_next(model, &iter);
+  }
+  if(found)
+    gtk_combo_box_set_active(GTK_COMBO_BOX(widget), i);
+}
+
+void dt_gui_preferences_enum_reset(GtkWidget *widget)
+{
+  const char *key = gtk_widget_get_name(widget);
+  const char *str = dt_confgen_get(key, DT_DEFAULT);
+  _gui_preferences_enum_set(widget, str);
+}
+
+static gboolean
+_gui_preferences_enum_reset(GtkWidget *label, GdkEventButton *event, GtkWidget *widget)
+{
+  if(event->type == GDK_2BUTTON_PRESS)
+  {
+    dt_gui_preferences_enum_reset(widget);
+    return TRUE;
+  }
+  return FALSE;
+}
+
+void dt_gui_preferences_enum_update(GtkWidget *widget)
+{
+  const char *key = gtk_widget_get_name(widget);
+  char *str = dt_conf_get_string(key);
+  _gui_preferences_enum_set(widget, str);
+  g_free(str);
+}
+
+GtkWidget *dt_gui_preferences_enum(GtkGrid *grid, const char *key)
+{
+  GtkWidget *w_label = gtk_label_new(_(dt_confgen_get_label(key)));
+  gtk_label_set_ellipsize(GTK_LABEL(w_label), PANGO_ELLIPSIZE_END);
+  gtk_widget_set_tooltip_text(w_label, _(dt_confgen_get_tooltip(key)));
+  gtk_widget_set_halign(w_label, GTK_ALIGN_START);
+  GtkWidget *labelev = gtk_event_box_new();
+  gtk_widget_set_hexpand(labelev, TRUE);
+  gtk_widget_add_events(labelev, GDK_BUTTON_PRESS_MASK);
+  gtk_container_add(GTK_CONTAINER(labelev), w_label);
+
+  GtkTreeIter iter;
+  GtkListStore *store = gtk_list_store_new(2, G_TYPE_STRING, G_TYPE_STRING);
+  gchar *str = dt_conf_get_string(key);
+  const char *values = dt_confgen_get(key, DT_VALUES);
+  gint i = 0;
+  gint pos = -1;
+  GList *vals = dt_util_str_to_glist("][", values);
+  for(GList *val = vals; val; val = g_list_next(val))
+  {
+    char *item = (char *)val->data;
+    // remove remaining [ or ]
+    if(item[0] == '[') item++;
+    else if(item[strlen(item) - 1] == ']') item[strlen(item) - 1] = '\0';
+    gtk_list_store_append(store, &iter);
+    // FIXME
```

Comment:
not obvious what is to be fixed here, can you add some more comment.

Reply:
copying bin/preferences_gen.h I was not able to make work:
` C_("preferences", item)`
so I've used 
`_(item)`
which works so far.

Do I explain this or do I remove the comment ?

Reply:
Because I suppose content of item is already translated somewhere else but not under the "preferences" context.

Reply:
Hummm.... My mistake should be elsewhere:

```
preferences.c:2428:68: error: expected ‘)’ before ‘item’
 2428 |     gtk_list_store_set(store, &iter, 0, item, 1, C_("preferences", item), -1);
```

Reply:
Meaning that macro` C_()` takes only strings as parameters. I have no solution to propose....

Reply:
OK. Thank you.